conans/errors.py

```diff
@@ -22,6 +22,9 @@ def conanfile_exception_formatter(conanfile_name, func_name):
     """
     try:
         yield
+    except ConanInvalidConfiguration as exc:
+        msg = "{}: Invalid configuration: {}".format(conanfile_name, exc)  # TODO: Move from here?
```

Comment:
I don't like having this message formatted here... any ideas?

Reply:
I think it is consistent to the way exceptions from conanfile.py are managed. Check ``_format_conanfile_exception()``, it might be worth to print the line number, etc. in the error message too.

Otherwise, if you wanted something specific to configure(), you could try-except the call to ``conanfile.configure()`` in ``graph_builder.py``, but wouldn't be much cleaner either.

Reply:
I prefer to handle the exception message here, better not pollute the graph code.

_This is not an error_, is it? so I don't think that adding information about file (it is always `conanfile.py`), or line (always inside `configure`) adds any information to the caller. It contains the **conan ref** and the **message defined in the recipe**, so it should be enough.

Reply:
I was thinking about lazy ``raise ConanInvalidConfiguration("Unsupported platform")``, and which some context, probably seen the line above could help. But yes, fair, leave it as is, it is good.